src/Cloud/weglide/UploadIGCFile.cpp

```diff
@@ -0,0 +1,241 @@
+  return UTF8ToWideConverter(value.is_string() ?
+    value.get_string().c_str() : error_code);
+}
+#else  //  _UNICODE
+static const std::string 
+GetJsonString(boost::json::standalone::value json_value, std::string_view key,
+              const char* error_code = "") noexcept {
+  auto value = json_value.at(key);
+  return value.is_string() ?
+    value.get_string().c_str() : error_code;
+}
+#endif  // _UNICODE
+
+namespace WeGlide {
+
+struct Glider {
+  uint64_t id = 0;
+  StaticString<0x40> name;
+  StaticString<4> kind;  // 'MG' - motor glider,...
+  StaticString<10> sc_class;
+};
+
+struct UploadResponse {
+  uint64_t flight_id = 0;
+  struct Pilot pilot;
+  struct Glider glider;
+  StaticString<0x40> scoring_date;
+  StaticString<0x40> registration;
+  StaticString<0x40> competition_id;
+};
+
+#if 0  // TODO this part is to delete if Codacy accept the pure function!?!
+// UpdateTask is without 'noexcept'!
+static Co::InvokeTask
+UpdateTask(Path igc_path, const char* uri, const struct Pilot& pilot,
+           uint_least32_t glider_id, boost::json::value& json,
+           ProgressListener& progress) {
+  try {
+    json = co_await UploadFlight(*Net::curl, uri, pilot, glider_id,
+      igc_path, progress);
+  }
+  catch (const std::exception& e) {
+    // Passing the exception without 'Co::InvokeTask' ???
+    throw std::runtime_error(e.what());
+  }
+}
+#else
+static Co::InvokeTask
+UpdateTask(Path igc_path, const char* uri, const struct Pilot& pilot,
+           uint_least32_t glider_id, boost::json::value& json,
+           ProgressListener& progress) {
+  json = co_await UploadFlight(*Net::curl, uri, pilot, glider_id,
+    igc_path, progress);
+}
+#endif
+
+static void
+UploadJsonInterpreter(const boost::json::value &json,
+                      struct UploadResponse& response) {
+  if (json.is_array() && (json.get_array().size() > 0)) {
+    auto flight_obj = json.get_array().at(0);
+    response.scoring_date = GetJsonString(flight_obj, "scoring_date").c_str();
+    response.flight_id = GetJsonInt(flight_obj, "id");
+    response.registration = GetJsonString(flight_obj, "registration").c_str();
+    response.competition_id = GetJsonString(flight_obj, "competition_id").c_str();
+    auto user = flight_obj.at("user");
+    if (user.is_object()) {
+      auto user_obj = user.as_object();
+      response.pilot.id = GetJsonInt(user_obj, "id");
+      response.pilot.name = GetJsonString(user_obj, "name").c_str();
+    }
+    auto aircraft = flight_obj.at("aircraft");
+    if (aircraft.is_object()) {
+      auto aircraft_obj = aircraft.as_object();
+      response.glider.id = GetJsonInt(aircraft_obj, "id");
+      response.glider.name = GetJsonString(aircraft_obj, "name").c_str();
+      response.glider.kind = GetJsonString(aircraft_obj, "kind").c_str();
+      response.glider.sc_class = GetJsonString(aircraft_obj, "sc_class").c_str();
+    }
+  }
+  else {
+    throw std::runtime_error("The http response has a wrong JSON format!");
```

Comment:
Unnecessary code. Just do `json.as_array()` and boost::json will check the type and throw an exception for you.